lib/internal/url.js

```diff
@@ -456,6 +159,352 @@ class URL {
   }
 }
 
+Object.defineProperties(URL.prototype, {
+  toString: {
+    // https://heycam.github.io/webidl/#es-stringifier
+    writable: true,
+    enumerable: true,
+    configurable: true,
+    value: function(options) {
```

Comment:
nit: I’d suggest naming this function (i.e. `function toString(options)`), otherwise V8 would infer its name as `value`